src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentReferenceCaptureLoweringPass.cs

```diff
@@ -33,32 +35,106 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
 
                 if (node.TagHelper.IsRefTagHelper())
                 {
-                    reference.Replace(RewriteUsage(@class, reference.Parent, node));
+                    // We've found an @ref directive attribute.
+                    //
+                    // If we can't get a nonempty identifier, do nothing because there will
+                    // already be a diagnostic for empty values
+                    var identifier = DetermineIdentifierToken(node);
+                    if (identifier == null)
+                    {
+                        continue;
+                    }
+
+                    var rewritten = RewriteUsage(reference.Parent, identifier);
+                    reference.Replace(rewritten);
+
+                    // Now we need to check if the field generation has been suppressed.
+                    //
+                    // You have to suppress field generation for generic types because we don't know the
+                    // type name to create the field.
+                    var generateField = ShouldGenerateField(reference.Parent);
```

Comment:
For clarity `var componentStartTag = reference.Parent;`

Reply:
I don't get this sorry.